joomla/joomla-cms
# Allow tracking of assets to be overriden for UCM table

Pull Request for Issue #8770 .

### Summary of Changes
When storing any item that uses tags, an entry will be created in the assets table even though tags doesn't use this asset. This creates a bogus entry in the assets table that doesn't belong there nor is it removed when the item is removed.

This PR introduces a setter to allow extensions to override the $_trackAssets setting, so that no bogus entries are created.

@sanderpotjer up your street :)

### Testing Instructions
1. Create a new article and make sure it has a tag assigned to the article
2. Store the article
3. Check the assets table. There is now an entry with a name in the form of #__ucm_content.9. The number will differ based on the entry in the ucm_content table.
4. Check the ucm_content table, in the row with the new article there will be an ID in the asset_id column that matches the last number in the name field of step 3.
5. Apply the patch
6. Create a new article and make sure it has a tag assigned to the article
7. Store the article
8. Check the assets table. No new entry has been added
9. Check the ucm_content table, in the row with the new article there will be no value in the asset_id column.

### Documentation Changes Required
None
